### PR TITLE
Add more accurate error message for WhoAmI command

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
- [x] Clarified the error if the selfsubjectreview API is not enabled in the cluster.
- [x] Add tests
- [x] Also react to `Forbidden` errors

#### Special notes for your reviewer:
Changes from this PR were proposed in the following comment
https://github.com/kubernetes/kubernetes/pull/111333#discussion_r980295949

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go [hunk 1]
```diff
@@ -139,16 +139,21 @@ func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams g
...
+
 // Run prints all user attributes.
 func (o WhoAmIOptions) Run() error {
 	sar := &authenticationv1alpha1.SelfSubjectReview{}
 	response, err := o.authClient.SelfSubjectReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return fmt.Errorf("the selfsubjectreviews API is not enabled in the cluster.\n" +
-				"enable APISelfSubjectReview feature gate and authentication.k8s.io/v1alpha1 API.")
+		switch {
```
[the remark this answers is not in the document]
Comment (nabokihms) [reply staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go:151]: I don't think this is important, yet I have no objections.

[the remark this answers is not in the document]
Comment (nabokihms) [reply staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go:151]: FIxed

File: staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go [hunk 2]
```diff
@@ -139,16 +139,21 @@ func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams g
 	return cmd
 }
 
+var notEnabledErr = fmt.Errorf(
+	"the selfsubjectreviews API is not enabled in the cluster or you do not have permission to call: " +
```
[the remark this answers is not in the document]
Comment (nabokihms) [reply staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go]: FIxed

File: staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (liggitt) [reply staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go]: I actually think the distinct messages are helpful... if we get a 404, we know the feature isn't available, so it's nice to not give them a potentially confusing message about not having permission in that case

[the remark this answers is not in the document]
Comment (nabokihms) [reply staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go:156]: Good point, fixed.

**Comment by sftim** [comment]
This looks like a change that users could notice. I recommend adding a release note.

**Comment by liggitt** [comment]
> This looks like a change that users could notice. I recommend adding a release note.

the whoami command is net new in 1.26, so I don't think a change note is required here

**Pull request merged**